sorc/chgres_cube.fd/surface.F90

```diff
@@ -164,26 +170,54 @@ subroutine surface_driver(localpet)
 !-----------------------------------------------------------------------
 
  if (convert_nst) call create_nst_esmf_fields
+ 
+!-----------------------------------------------------------------------
+! Adjust soil levels of input grid !! not implemented yet
+!-----------------------------------------------------------------------
+
+ call adjust_soil_levels(localpet)
 
 !-----------------------------------------------------------------------
 ! Horizontally interpolate fields.
 !-----------------------------------------------------------------------
 
  call interp(localpet)
-
+ 
+ if (.not. (vgtyp_from_climo .or. sotyp_from_climo)) then
+   !---------------------------------------------------------------------------------------------
+   ! Check for points where smois is too high to be a land point at a land point
+   !---------------------------------------------------------------------------------------------
+   !  
+   !call check_smois_water
+ endif
+ 
 !---------------------------------------------------------------------------------------------
-! Adjust soil/landice column temperatures for any change in elevation between the
+! Adjust soil/landice column temperatures for any change in elevation between
+! the
 ! input and target grids.
 !---------------------------------------------------------------------------------------------
 
  call adjust_soilt_for_terrain
-
+ 
 !---------------------------------------------------------------------------------------------
 ! Rescale soil moisture for changes in soil type between the input and target grids.
 !---------------------------------------------------------------------------------------------
 
  call rescale_soil_moisture
+ 
+ if (.not. (vgtyp_from_climo .or. sotyp_from_climo)) then
+  !---------------------------------------------------------------------------------------------
+  ! Check soil moisture again for mismatches after rescale_soil_moisture subroutine
```

Comment:
Why are there mismatches?  Do they first appear after the call to 'rescale_soil_moisture', or earlier?